src/wasm/asm-wasm-builder.cc

```diff
@@ -453,26 +574,108 @@ class AsmWasmBuilderImpl : public AstVisitor {
   }                                                                   \
   break
 
+  Expression* GetLeft(BinaryOperation* expr) {
```

Comment:
0|x is not a valid coercion. Only x|0 is. As for the others, they also necessarily convert so that non trivial part is on left.